### DISCUSSION TITLE
Different Render Settings Override for each blend file in Loom Batch

### DISCUSSION THREAD

**Discussion by DrSakuu** [p2or/blender-loom#63] [Ideas]
Right now you can override the render settings with a preset, but this is applied to all blend files in a batch. I render posters in Blender, and some of them are landscape and some are portrait resolution. If there was a button to set a preset for each blend file individually in the Loom Batch window, I could render all the posters in one go.

**Comment by p2or** [comment]
Hi @DrSakuu,

thanks for your feature request. To get an understanding of this: Why can't you just create a preset without the actual resolution? Or, for example, apply the preset, save a new version of that file and add this one to the queue? 

In general this should be easy to implement. The problem I personally see and the remaining question is: where to put the preset setting within each row without bloating the UI and destroying the current logic...? If you have a good idea or mockup to solve this, please let me know, I would appreciate your feedback.

Cheers,
Christian

**Comment by DrSakuu** [comment]
Yes, those are the same exact thoughts I had. I don't have a plan how to implement this nicely, but saving a separate blend file is a temporary solution. This can wait until you can make room in the interface.

**Comment by p2or** [comment]
Hi @DrSakuu,

had some time and implemented that the scene, the camera and the render preset can be overridden per item in the queue. Still not satisfied with the UI though but if you have time, I would be pleased if you could try it out and report back:

https://github.com/p2or/blender-loom/assets/512368/2439d0ef-7901-4039-bc7c-b5a4837affe9

If you are interessted, you can download the [latest changes from the corresponding branch](https://github.com/p2or/blender-loom/tree/batch-overrides-ui) and install *Loom* as usual.

Cheers,
Christian

**Comment by DrSakuu** [comment]
Thanks, this layout works well for me, but the interface is very unresponsive when you add a complex blend file. Every input takes like a second to register, and it gets worse and worse with more blend files in the list. This does not happen with 0.9.3 release.

**Comment by p2or** [comment]
Hi @DrSakuu,

thanks for taking the time to test the branch. I will try to optimize adding larger files and let you know when I'm done. 

Cheers,
Christian

**Comment by p2or** [comment]
Hi @DrSakuu,

If you haven't noticed, I [updated the feature branch](https://github.com/p2or/blender-loom/tree/batch-overrides-ui) a few weeks ago to speed up the camera and scene selection within the dialog itself. Also, the latest changes from the master branch are now included.

However, I haven't had the time to test it extensively. If you have the time, please download the [latest version](https://github.com/p2or/blender-loom/archive/refs/heads/batch-overrides-ui.zip) and let me know your thoughts so we can hopefully merge it in the upcoming weeks.

Cheers,
Christian

**Comment by DrSakuu** [comment]
Thanks, it is working. Here are some things I found: 

- It is unintuitive that the drop down for presets only appears after you make presets, it is hard to discover the feature without the readme. 
- When trying to save a preset with an existing name, it should ask if you want to update the preset instead of just failing with an error. 
- When you clone a file in the batch render window, it doesn't clone the override preset. 

Slightly offtopic: 

- The buttons for cloning and removing files in the batch have incorrect tooltips.
- When you hit ok to start the command line batch render, it should save the blend file so that your changes actually have an effect.

**Comment by p2or** [comment]
Hi @DrSakuu,

thank you for your feedback and for taking the time to test the branch.

> - It is unintuitive that the drop down for presets only appears after you make presets, it is hard to discover the feature without the readme.
> - When trying to save a preset with an existing name, it should ask if you want to update the preset instead of just failing with an error.

That's the preset system that comes with Blender itself and I don't see anything I can do about it besides implementing a custom one, (see e.g. the camera presets for reference).

> - When you clone a file in the batch render window, it doesn't clone the override preset.

Good catch, I'll add a check for that.

> - The buttons for cloning and removing files in the batch have incorrect tooltips.

Thanks.

> - When you hit ok to start the command line batch render, it should save the blend file so that your changes actually have an effect.

When rendering each file, Loom changes the actual settings on the fly so I don't get why saving the blend is relevant here. Would you mind elaborating? Doesn't the file render correctly?

If something else comes to your mind, please let me know.

Cheers,
Christian

**Reply by DrSakuu** [reply]
> When rendering each file, Loom changes the actual settings on the fly so I don't get why saving the blend is relevant here. Would you mind elaborating? Doesn't the file render correctly?

When using the Loom Batch dialog to render an image sequence, it starts command line Blender in a new window and opens the blend file externally, so if you have unsaved changes, those won't be in the blend file. When rendering in the command line using the single Render Image Sequence dialog, the blend file is saved before the command line window is started. The same thing needs to happen when using the batch render dialog.
